File: service/cms-service/src/main/java/dev/xbase/controller/api/auth/AuthAPI.java
package dev.xbase.controller.api.auth;

import dev.xbase.controller.api.auth.model.JwtResponse;
import dev.xbase.controller.api.auth.model.LoginRequest;
import dev.xbase.controller.api.auth.model.TokenRefreshRequest;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("/pub/authenticate")
public interface AuthAPI {
    @PostMapping
    JwtResponse auth(@Valid @RequestBody LoginRequest loginRequest);

    @PostMapping("/refresh-token")
    JwtResponse refreshToken(@Valid @RequestBody TokenRefreshRequest request);
}

File: ui/src/services/authApi.ts
import http from '@/utils/api/api.ts';
import { LoginResult } from '@/pages/authentication/login/data';

export const authApi = {

  async refreshToken(params: { refreshToken: string }): Promise<LoginResult> {
    const data = await http.post<LoginResult>("/pub/authenticate/refresh-token", {
      refreshToken: params.refreshToken
    });
    return data.data
  },
};
